app/gui2/src/stores/project/index.ts

```diff
@@ -53,12 +53,11 @@ function resolveLsUrl(config: GuiConfig): LsUrls {
     const dataUrl = engine.dataUrl
     const rpcUrl = engine.rpcUrl
     let ydocUrl
-    if (engine.ydocUrl == null || engine.ydocUrl === '') {
+    if (engine.ydocUrl === 'undefined') {
```

Comment:
Really `'undefined'` as a string? Don't you want to `if (!engine.ydocUrl)`? Or `typeof engine.ydocUrl === 'undefined'`?

Reply:
Yes, it's a special value because config values can only be strings.

Reply:
The config values can also be left missing, making them actually undefined here. We still should support that case.

I'm fine another or case here with a comment with why this is necessary, if it is needed for any particular reason.  But ideally i'd not modify this condition, as stringified `undefined` is often a trap that's hard to spot during debugging. In any case, the previous conditions should still be preserved.

```suggestion
    if (engine.ydocUrl == null || engine.ydocUrl === '' || engine.ydocUrl === 'undefined') {
```